Weekly Release patch fails to apply when dev diverges from main in version-bumped files

## Summary

The Weekly Release workflow generates a version-bump patch relative to `main`, but `safe_outputs` applies it onto `dev`. When files touched by the patch (`CHANGELOG.md`, `SKILL.md`) have diverged between `main` and `dev`, `git am` fails with context mismatch errors.

## Evidence

[Run 23418394262](https://github.com/ahmadabdalla/azure-cost-calculator/actions/runs/23418394262) (2026-03-23) completed the agent phase successfully (produced a valid `v1.5.2` patch), but the `safe_outputs` job failed:

```
error: patch failed: CHANGELOG.md:20
error: patch failed: skills/azure-cost-calculator/SKILL.md:6
##[error]✗ Message 1 (create_pull_request) failed: Failed to apply patch
```

## Root cause

1. The prompt (Step 5a) instructs the agent to `git checkout origin/dev -- .claude-plugin/plugin.json CHANGELOG.md skills/azure-cost-calculator/SKILL.md` while checked out on `main`.
2. The agent modifies those files and commits. The resulting `git format-patch` encodes context lines from `main` (the base), not `dev`.
3. `safe_outputs` checks out `dev`, creates a branch from `dev`, and runs `git am` with the patch.
4. PR #575 (`chore: codebase documentation clean up and standardisation`) changed `CHANGELOG.md` and `SKILL.md` on `dev` (em-dash to semicolon/colon replacements). Those same lines appear as unchanged context in the patch, but they no longer match `dev`'s content, so `git am` rejects the patch.

The core issue: the patch carries `main`-relative context but is applied to a `dev` checkout. Any time `dev` diverges from `main` in the same files the agent modifies, this will fail.

## Affected workflow

- `.github/workflows/weekly-release.lock.yml` (compiled from `weekly-release.md`)
- Prompt Steps 5a through 5e (version file import and patch generation)

## Reproduction

Triggers whenever:
1. A PR merged to `dev` modifies lines in `CHANGELOG.md` or `SKILL.md` that also appear as context in the version-bump patch
2. Those changes have not yet been released (merged back to `main`)

## Possible fixes

- Have the agent generate the patch relative to `dev` (not `main`), so context lines match the `dev` checkout where the patch is applied.
- Alternatively, have `safe_outputs` apply with `git am --3way` or `git apply --3way` to allow fuzzy matching when context diverges.

--- a/.github/workflows/weekly-release.md
+++ b/.github/workflows/weekly-release.md
@@ -125,7 +125,7 @@ The agent is checked out on the default branch, which may differ from `dev`. Imp
 git checkout origin/dev -- .claude-plugin/plugin.json CHANGELOG.md skills/azure-cost-calculator/SKILL.md
 ```
 
-This ensures you are editing the `dev` versions of these files. The patch context lines for each file are deliberately placed around stable structural markers (`<!-- versions -->` in CHANGELOG.md, frontmatter `metadata:` block at the top of SKILL.md, JSON `{` in plugin.json) to minimise the chance of context mismatch when `safe-outputs` applies the patch on `dev`.
+This ensures you are editing the `dev` versions of these files. The resulting patch will apply cleanly when `safe-outputs` applies it on `dev`.
 
 ### 5b. Update `CHANGELOG.md`
 
@@ -165,7 +165,7 @@ Update the `"version"` field to the new version.
 
 ### 5d. Update `SKILL.md`
 
-Update the `version:` field in the YAML frontmatter of `skills/azure-cost-calculator/SKILL.md` to the new version. The field is at the top of the frontmatter under `metadata:` (line 4), ensuring its patch context lines are always stable (`---`, `metadata:`, `  author:`).
+Update the `version:` field in the YAML frontmatter of `skills/azure-cost-calculator/SKILL.md` to the new version.
 
 ### 5e. Commit the version bump
 

--- a/skills/azure-cost-calculator/SKILL.md
+++ b/skills/azure-cost-calculator/SKILL.md
@@ -1,12 +1,12 @@
 ---
-metadata:
-  author: ahmadabdalla
-  version: "1.5.1"
 name: azure-cost-calculator
 description: Helps estimate and calculate Azure resource costs. Use this skill when users ask about Azure pricing, cost estimation, resource sizing costs, comparing pricing tiers, budgeting for Azure deployments, or understanding Azure billing. Triggers include questions like "how much will this cost in Azure", "estimate Azure costs", "compare Azure pricing", "budget for Azure resources".
 license: MIT
 argument-hint: "<azure-service-name>"
 compatibility: Requires curl + jq (macOS/Linux) or PowerShell 7+ (pwsh) or Windows PowerShell 5.1 (powershell.exe on Windows), and internet access to prices.azure.com. No Azure subscription needed.
+metadata:
+  author: ahmadabdalla
+  version: "1.5.1"
 ---
 
 # Azure Cost Calculator

